src/openassetio-core/include/openassetio/access.hpp

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <array>
+
+#include <openassetio/export.h>
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+/**
+ * Access modes available for API operations.
+ */
+namespace access {
+
+/**
+ * @name Access Pattern
+ */
+
+/**
+ * Common constant values for strong enumerations.
+ *
+ * We must ensure correspondence between the values of the various
+ * workflows' strong enums. This allows for comparison, string lookup,
+ * simplifies serialisation, and supports language (especially C)
+ * bindings.
+ */
+enum Access { kRead = 0, kWrite, kCreateRelated };
```

Comment:
[question] In `errors.h` we use `#define`s for constants to make them useable in C.  But we could/should have used a C enum instead, like here, I think(?).

Given the C interface is on the back burner, I didn't bother extracting this enum out to a C-compatible `.h` file.  Is that worth doing now?